core-backend/src/funcs.rs

```diff
@@ -128,13 +85,13 @@ impl From<u32> for SyscallReturnValue {
     }
 }
 
-pub(crate) trait SyscallContext: Sized {
-    fn from_args(args: &[Value]) -> Result<(Self, &[Value]), HostError>;
+pub(crate) trait SyscallContext: Sized + Copy {
+    fn from_args(args: &[Value]) -> Result<(Self, Option<Gas>, &[Value]), HostError>;
```

Comment:
why is gas optional?

Reply:
Since we have an `impl SyscallContext for ()` that doesn't have gas

Reply:
seems we could remove it? @ark0f ?